torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -39,25 +39,25 @@ using namespace torch::autograd::generated;
 
 namespace torch { namespace autograd {
 
-inline void check_inplace(const Tensor& tensor) {
+inline void check_inplace(const Tensor& tensor, bool requires_grad) {
   auto& var = static_cast<const Variable&>(tensor);
-  if (var.requires_grad() && GradMode::is_enabled()) {
+  if (requires_grad && GradMode::is_enabled()) {
     if (var.is_view()) {
       // NB: is_view() ==> get_autograd_meta()
       auto diff_view_meta = static_cast<DifferentiableViewMeta*>(impl::get_autograd_meta(var));
       // This can throw or warn
       handle_view_on_rebase(diff_view_meta);
     }
-    if (var.is_leaf()) {
+    if (tensor.requires_grad() && var.is_leaf()) {
```

Comment:
I know it doesn't matter but seeing a test for requires grad on tensor and is leaf on var makes go "wait what" (they're the same thing)

Reply:
Ho I didn't realize. Let me remove the var altogether.